src/runtime/security/negotiation_manager.cpp

```diff
@@ -38,80 +40,100 @@ inline bool in_white_list(task_code code)
     return is_negotiation_message(code) || fd::is_failure_detector_message(code);
 }
 
-negotiation_map negotiation_service::_negotiations;
-utils::rw_lock_nr negotiation_service::_lock;
+negotiation_map negotiation_manager::_negotiations;
+utils::rw_lock_nr negotiation_manager::_lock;
 
-negotiation_service::negotiation_service() : serverlet("negotiation_service") {}
+negotiation_manager::negotiation_manager() : serverlet("negotiation_manager") {}
 
-void negotiation_service::open_service()
+void negotiation_manager::open_service()
 {
     register_rpc_handler_with_rpc_holder(
-        RPC_NEGOTIATION, "Negotiation", &negotiation_service::on_negotiation_request);
+        RPC_NEGOTIATION, "Negotiation", &negotiation_manager::on_negotiation_request);
 }
 
-void negotiation_service::on_negotiation_request(negotiation_rpc rpc)
+void negotiation_manager::on_negotiation_request(negotiation_rpc rpc)
 {
     dassert(!rpc.dsn_request()->io_session->is_client(),
-            "only server session receive negotiation request");
+            "only server session receives negotiation request");
 
     // reply SASL_AUTH_DISABLE if auth is not enable
     if (!security::FLAGS_enable_auth) {
         rpc.response().status = negotiation_status::type::SASL_AUTH_DISABLE;
         return;
     }
 
-    server_negotiation *srv_negotiation = nullptr;
-    {
-        utils::auto_read_lock l(_lock);
-        srv_negotiation =
-            static_cast<server_negotiation *>(_negotiations[rpc.dsn_request()->io_session].get());
+    std::shared_ptr<negotiation> nego = get_negotiation(rpc);
+    if (nullptr != nego) {
+        server_negotiation *srv_negotiation = static_cast<server_negotiation *>(nego.get());
+        srv_negotiation->handle_request(rpc);
     }
+}
```

Comment:
Is it necessary to add error log even error handling for nego is nullptr? The same in on_negotiation_response function.

Reply:
I think it's ok to add a log here. And there won't be too much log here, because it doesn't happen very often

.